test/06-log-management.bats

```diff
@@ -0,0 +1,230 @@
+    [ -f "$LOG_PATH" ]
+
+    # Check file permissions (should be 0640)
+    local perms
+    perms=$(stat -c "%a" "$LOG_PATH")
+    [ "$perms" = "640" ]
+}
+
+@test "log management: should reject unsafe log paths" {
+    # Test with symlink path (if we can create one safely)
+    local test_dir="$TEST_TMPDIR/conmon-test-symlink-$$"
+    if ! mkdir -p "$test_dir/real" 2>/dev/null; then
+        skip "Cannot create test directory for symlink test"
+    fi
+
+    if ! ln -s "$test_dir/real" "$test_dir/link" 2>/dev/null; then
+        rm -rf "$test_dir"
+        skip "Cannot create symlink for test"
+    fi
+
+    local symlink_log="$test_dir/link/test.log"
+
+    run_conmon --cid "$CTR_ID" --cuuid "$CTR_ID" --runtime "$VALID_PATH" \
+        --log-path "k8s-file:$symlink_log" --log-rotate --log-max-files 2
+
+    # Should either fail or succeed but handle safely
+    [[ "$status" -eq 0 || "$status" -eq 1 ]]
+
+    # Cleanup
+    rm -rf "$test_dir"
+}
+
+# === Backward Compatibility Tests ===
+
+@test "log management: should maintain backward compatibility" {
+    # Test that old truncation behavior still works
+    run_conmon_k8s_log --log-size-max 1024
+    assert_success
+    [ -f "$LOG_PATH" ]
+
+    # Test that new options don't break when not used
+    run_conmon_k8s_log
+    assert_success
+    [ -f "$LOG_PATH" ]
+}
+
+# === Error Recovery Tests ===
+
+@test "log management: should handle file system errors gracefully" {
+    # Skip this test if running as root (root can bypass permission restrictions)
+    if [[ $EUID -eq 0 ]]; then
+        skip "Skipping permission test when running as root"
+    fi
+
+    # Test with read-only directory (if possible)
+    local readonly_dir="$TEST_TMPDIR/conmon-readonly-$$"
+    if ! mkdir -p "$readonly_dir" 2>/dev/null; then
+        skip "Cannot create test directory for readonly test"
+    fi
+
+    # Make directory read-only
+    if ! chmod 444 "$readonly_dir" 2>/dev/null; then
+        rmdir "$readonly_dir"
+        skip "Cannot make directory readonly for test"
+    fi
+
+    local readonly_log="$readonly_dir/test.log"
+
+    run_conmon --cid "$CTR_ID" --cuuid "$CTR_ID" --runtime "$VALID_PATH" \
+        --log-path "k8s-file:$readonly_log" --log-rotate --log-max-files 2
+
+    # Should fail gracefully without crashing
+    assert_failure
+
+    # Cleanup
+    chmod 755 "$readonly_dir" 2>/dev/null || true
+    rmdir "$readonly_dir" 2>/dev/null || true
+}
+
+# === Allowlist Functionality Tests ===
+
+@test "log management: should work with log allowlist" {
+    # Test allowlist functionality if configured
+    local allowed_dir="$TEST_TMPDIR/conmon-allowed-$$"
+    if ! mkdir -p "$allowed_dir" 2>/dev/null; then
+        skip "Cannot create test directory for allowlist test"
+    fi
+
+    local allowed_log="$allowed_dir/test.log"
+
+    run_conmon --cid "$CTR_ID" --cuuid "$CTR_ID" --runtime "$VALID_PATH" \
+        --log-path "k8s-file:$allowed_log" \
+        --log-allowlist-dirs "$allowed_dir" \
```

Comment:
Can you also test second `--log-allowlist-dirs`?

```
--log-allowlist-dirs "$allowed_dir"
--log-allowlist-dirs /foo
```

So we know that multiple options are enabled and /foo does not override the first one.